### PR TITLE
CI: move all tests in a single workflow 

### PR DESCRIPTION
**Pull request by nicoloboschi**
Descriptions of the changes in this PR:

Currently each CI job has his own workflow which makes hard to understand the whole picture of a CI execution.

I've worked on a new workflow that includes all the existing tests and verification. The steps:
1. Validate the pull request -> compile, style, licenses, javadoc and so on
If step 1 passes, trigger all the tests and builds (windows, macos)

This is an example of the result on my fork: https://github.com/nicoloboschi/bookkeeper/actions/runs/3894518085

I kept the same job names because we have to keep the same required jobs to pass in order to merge a pull request.
In case of failures, it's possible to trigger the failed jobs manually via the Github UI or using `rerun failure checks`

### CODE REVIEW

**Review by hangc0276 — APPROVED** [review]

**Comment by hangc0276** [comment]
@yaalsn Would you please help take a look at this PR, thanks.

**Review by yaalsn — APPROVED** [review]
LGTM

File: .github/workflows/bk-ci.yml [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+      - name: print JVM thread dumps when cancelled
+        if: cancelled()
+        run: ./dev/ci-tool print_thread_dumps
+
+  owasp-dependency-check:
+    name: OWASP Dependency Check
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    needs: [ 'build-and-license-check' ]
+    if: ${{ needs.build-and-license-check.outputs.need_owasp == 'true' }}
```
Comment (yaalsn) [inline .github/workflows/bk-ci.yml:385]: A question: Do we need to skip this job when the change belongs to `docs` in `.github/changes-filter.yaml` file ?

File: .github/workflows/bk-ci.yml [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+        # CVEs that we cannot patch up anyways
+        run: mvn -q -B -ntp clean install verify -Powasp-dependency-check -DskipTests -pl '!stream/distributedlog/io/dlfs'
+
+      - name: Upload report
+        uses: actions/upload-artifact@v3
+        if: ${{ cancelled() || failure() }}
+        continue-on-error: true
+        with:
+          name: dependency report
+          path: target/dependency-check-report.html
```
Comment (yaalsn) [inline .github/workflows/bk-ci.yml:419]: Do we need to add `retention-days: 7` for this job?

**Comment by nicoloboschi** [comment]
Thanks @yaalsn, I've added the retention. The default value is 90 days and that's really a waste of resources (even thought is just a html static page 😄 )

**Pull request merged**